### ISSUE TITLE
Parameter names imported, but want parameter descriptions?

### ISSUE THREAD

**Issue by JimboMahoney** [RGLab/flowCore#154]
**Describe the bug**
Hi Mike,

I don't think this is a "bug", but I'm wondering how to import the parameter "descriptions" rather than their names.

Bear in mind I'm new to both flow/mass cytometry and R programming...

I've tried to Google before raising an issue, but can't find an answer.

**To Reproduce**
I'm reading an FCS file generated on a Helios mass cytometer:

raw_fcs<-read.FCS("CellLine1_00.fcs", alter.names = TRUE)

I then convert to a data frame:

FCSDATA <- as.data.frame(exprs(raw_fcs))

This is great and it gives me all the data.

However, the parameter names (columns) are named like this:

Nd143Di / Nd144Di etc.

When I open the FCS in e.g. FCS Express or FlowJo, it seems they are reading a different section of the FCS file, which I can also see in e.g. LDATAPP. They appear like this:

143Nd_GITR / 144Nd_Blank1 etc.

e.g. they contain the extra detail that has been added by the user during acquisition.

This data appears in LDATAPP in the "S" column, rather than the "N" column as the data that read.FCS is currently importing as the parameter names.

How can I get read.FCS to import the second, more user-friendly parameter names?

Many thanks!

**Comment by jacobpwagner** [comment]
Hi Jimbo. Have you checked out the `markernames` method for the `flowFrame` class? This will return the $PnS values for your sample (the "S" column that is being written in to the FCS). Note that that is a method for a `flowFrame`, so you should call it on the object returned from `read.FCS` (`raw_fcs` in your example above) rather than converting it with `as.data.frame`.

**Comment by jacobpwagner** [comment]
To clarify, using your example above, you could access the longer descriptive names as follows:
```
fr <- read.FCS("CellLine1_00.fcs", alter.names = TRUE)
markernames(fr)
```
The default behavior there will be to omit the scatter and time channels, but you could also use the `parameters` method to get everything:
```
fr <- read.FCS("CellLine1_00.fcs", alter.names = TRUE)
parameters(fr)[["desc"]]
```

**Comment by JimboMahoney** [comment]
Thanks Jacob!

That got me on the right track. 

Sadly, it returns "NA" for a lot of columns from a CyTOF (e.g. Time, Event_Length, Offset etc.), so I had to figure out how to ONLY replace the columns in the FCS Data Frame that are NOT NA in the parameters list.

I did it like this:

1) Create list of descriptions from raw fcs file:

```
params<-parameters(raw_fcs)[["desc"]]
```

2) Replace only those columns in the data frame that are NOT NA in the descriptions list
```
colnames(FCSDATA)[!is.na(params)] <- na.omit(params)
```
